packages/user-group-service/src/groups/resolver.ts

```diff
@@ -39,17 +39,35 @@ export const GroupResolver = {
   Mutation: {
     addGroup ( root: any, { payload }: GraphQLArgs, ctx: any ) {
       return new Groups( payload )
-        .save();
+        .save()
+        .then( ( res: any ) => {
+          const transformedData = UserGroupAPIHelper.formatSearchInput( res );
+          UserGroupAPIHelper.manageSearchIndex(transformedData, 'index');
+          return res;
+        });
     },
     updateGroup ( root: any, { id, payload }: GraphQLArgs, ctx: any ) {
       return Groups
         .findByIdAndUpdate( id, { ...payload, updatedOn: new Date() }, { new: true } )
-        .exec();
+        .exec()
+        .then( ( res: any ) => {
+          const transformedData = UserGroupAPIHelper.formatSearchInput( res );
+          UserGroupAPIHelper.manageSearchIndex(transformedData, 'index');
+          return res;
+        });
     },
     deleteGroup ( root: any, { id }: GraphQLArgs, ctx: any ) {
       return Groups
         .findByIdAndRemove( id )
-        .exec();
+        .exec()
+        .then( ( res: any ) => {
+          const input = {
+            dataSource: "oneportal",
```

Comment:
We should probably use an environment variable for the data source.

Reply:
@riginoommen what do you think about it?

Reply:
I think we can use a global environment variable. I will be good also. I agree with it.